src/services/selfHosted/useSelfHostedUserList.ts

```diff
@@ -0,0 +1,82 @@
+import { useInfiniteQuery } from '@tanstack/react-query'
+import { useMemo } from 'react'
+import { z } from 'zod'
+
+import Api from 'shared/api'
+import { NetworkErrorObject } from 'shared/api/helpers'
+
+const OwnerSchema = z.object({
+  ownerid: z.number(),
+  username: z.string().nullable(),
+  email: z.string().nullable(),
+  name: z.string().nullable(),
+  isAdmin: z.boolean(),
+  activated: z.boolean(),
+})
+
+export type UserListOwner = z.infer<typeof OwnerSchema>
+
+const RequestSchema = z.object({
+  count: z.number(),
+  next: z.string().nullable(),
+  previous: z.string().nullable(),
+  results: z.array(OwnerSchema),
+  totalPages: z.number(),
+})
+
+interface UseSelfHostedUserListArgs {
+  activated?: boolean
+  search?: string
+  isAdmin?: boolean
+}
+
+export const useSelfHostedUserList = ({
+  activated,
+  search,
+  isAdmin,
+}: UseSelfHostedUserListArgs) => {
+  const { data, ...rest } = useInfiniteQuery({
+    queryKey: ['SelfHostedUserList', activated, search, isAdmin],
+    queryFn: ({ pageParam = 1, signal }) => {
+      return Api.get({
+        path: '/users',
+        signal,
+        query: {
+          activated,
+          isAdmin,
+          search: search ? search : undefined,
+          page: pageParam,
+        },
+      }).then((res) => {
+        const parsedData = RequestSchema.safeParse(res)
+
+        if (!parsedData.success) {
+          return Promise.reject({
+            status: 404,
+            data: {},
+            dev: 'useSelfHostedUserList - 404 schema parsing failed',
+          } satisfies NetworkErrorObject)
+        }
+        return parsedData.data
+      })
+    },
+    suspense: false,
+    getNextPageParam: (data) => {
+      if (data?.next) {
+        const { searchParams } = new URL(data.next)
+        return searchParams.get('page')
+      }
+      return undefined
+    },
+  })
+
+  const memoedData = useMemo(
+    () => data?.pages?.flatMap((page) => page.results) ?? [],
+    [data]
+  )
+
+  return {
+    data: memoedData,
+    ...rest,
```

Comment:
any chance we can get rid of the ...rest? Personally, think this would be the best opportunity to remove as much ambiguity as possible and from what I can see in other parts of the code base, we look to be destructuring isFetching, hasNextPage, fetchNextPage from there.

Reply:
I prefer to keep this as is, as it makes using the hook feel very similar to when using a query directly, also if someone is going to access values from it, they don't need to change the hook at all, they already have all the fields that they need.

Reply:
Yeah I gotcha there, makes sense. I can get behind that